pilot/pkg/config/kube/ingress/conversion.go

```diff
@@ -324,6 +324,30 @@ func shouldProcessIngressWithClass(mesh *meshconfig.MeshConfig, ingress *v1beta1
 	}
 }
 
+// TODO: WRITE TEST, add error handling
+func ShouldCreateGateway(ingress v1beta1.Ingress, domainSuffix string) *config.Config {
+	if istioGateway, exists := ingress.Annotations[kube.IngressIstioGatewayAnnotation]; exists {
+		gateway := &networking.Gateway{}
```

Comment:
I am not quite sure why we are creating any empty gateway? Won't that do nothing?

Reply:
Please check my comment here: https://github.com/istio/istio/issues/27205#issuecomment-692735668
The idea is to allow specifying a gateway, so this way we can have only one - that should solve the TLS / same certificate issue.

This is my first PR and I am new to the Istio code base. Sorry if there is anything confusing.

Reply:
I would have expected that we just do not create a gateway here at all though. Otherwise, won't we have multiple Gateways with the same name/namespace created? The "real" one, created by the user in Kubernetes, and all of these empty ones?

Reply:
I see what you mean. updated the PR.